File: backend/components/metiers/voice/fixtures/core/validators/fixtures_validator.py
"""
fixtures_validator.py - Validation avancée des fixtures voice (Python)
"""


def is_valid_voice_model(model):
    """Validate voice model fixture.

    Args:
        model (dict): The voice model fixture to validate.

    Returns:
        bool: True if the fixture is valid, False otherwise.
    """
    return bool(
        model
        and isinstance(model.get("id"), str)
        and isinstance(model.get("vertices"), list)
        and isinstance(model.get("faces"), int)
    )


def is_valid_user(user):
    """Validate user fixture.

    Args:
        user (dict): The user fixture to validate.

    Returns:
        bool: True if the fixture is valid, False otherwise.
    """
    return bool(
        user
        and isinstance(user.get("id"), str)
        and isinstance(user.get("role"), str)
    )

File: backend/components/metiers/mobile/fixtures/core/validators/fixtures_validator.js
// fixtures_validator.js - Validation avancée des fixtures Mobile

module.exports = {
  isValidmobileModel: (model) => {
    return model && typeof model.id === 'string' && Array.isArray(model.vertices) && Array.isArray(model.faces);
  },
  isValidUser: (user) => {
    return user && typeof user.id === 'string' && typeof user.role === 'string';
  }
};
